src/test/scala/nl.knaw.dans.easy.deposit/SubmitterSpec.scala

```diff
@@ -97,41 +118,41 @@ class SubmitterSpec extends TestSupportFixture with MockFactory {
   it should "report a file missing in the draft" in {
     val depositDir = createDeposit(datasetMetadata)
     val bag = getBag(depositDir)
-    (bag.baseDir.parent / "deposit.properties").append(s"identifier.doi=$doi")
+    addDoiToDepositProperties(bag)
     bag.addPayloadFile("lorum ipsum".inputStream, Paths.get("file.txt"))
     bag.save()
-    (testDir / "submitted").createDirectories()
 
     // add file to manifest that does not exist
     (bag.baseDir / "manifest-sha1.txt").append("chk file")
 
     assume(DDM.triedSchema.isAvailable)
-    new Submitter(testDir / "staged", testDir / "submitted").submit(depositDir) should matchPattern {
+    new Submitter(testDir / "stage-for-submit", testDir / "submitted", userGroup).submit(depositDir) should matchPattern {
       case Failure(e) if e.getMessage == s"invalid bag, missing [files, checksums]: [Set($testDir/drafts/user/${ depositDir.id }/bag/file), Set()]" =>
     }
   }
 
   it should "report an invalid checksum" in {
     val depositDir = createDeposit(datasetMetadata)
     val bag = getBag(depositDir)
-    (bag.baseDir.parent / "deposit.properties").append(s"identifier.doi=$doi")
+    addDoiToDepositProperties(bag)
     bag.addPayloadFile("lorum ipsum".inputStream, Paths.get("file.txt"))
     bag.save()
-    (testDir / "submitted").createDirectories()
 
     // change a checksum in the manifest
     val manifest = bag.baseDir / "manifest-sha1.txt"
     manifest.write(manifest.contentAsString.replaceAll(" +", "xxx  "))
 
     val checksum = "a57ec0c3239f30b29f1e9270581be50a70c74c04"
     assume(DDM.triedSchema.isAvailable)
-    new Submitter(testDir / "staged", testDir / "submitted").submit(depositDir) should matchPattern {
+    new Submitter(testDir / "stage-for-submit", testDir / "submitted", userGroup).submit(depositDir) should matchPattern {
       case Failure(e)
         if e.getMessage == s"staged and draft bag [${ bag.baseDir.parent }] have different payload manifest elements: (Set((data/file.txt,$checksum)),Set((data/file.txt,${ checksum }xxx)))" =>
     }
   }
 
-  private def getBagDir(depositDir: DepositDir) = getBag(depositDir).baseDir
+  private def addDoiToDepositProperties(bag: DansBag): File = {
```

Comment:
Yes, according to `better-files` it returns `File`, but since this method _appends_ to a file, you should return `Unit`.

```suggestion
  private def addDoiToDepositProperties(bag: DansBag): Unit = {
```